### PR TITLE
Remove deprecated es.max-num-spans

### LINKED ISSUE #2457: Remove deprecated `--es.max-num-spans` flag in v1.21.0
## Requirement - what kind of business use case are you trying to solve?

As a follow-up task to #2437, which deprecated `--es.max-num-spans` in favour of `--es.max-doc-size`, this is a reminder Issue to remove the deprecated `--es.max-num-spans` flag in version 1.21.0, as noted in the [deprecation note on the flag](https://github.com/jaegertracing/jaeger/pull/2453/files/9ebf9d37a2662f12696c4664d043d5b6c61eaebe#diff-33c1785172c58663ab0bc033adbcef49R182).

### PR DESCRIPTION
**Pull request by BernardTolosajr**
## Which problem is this PR solving?
- Remove deprecated `--es.max-num-spans` flag in v1.21.0 - resolves #2457
- Fixes part of #2738

## Short description of the changes
- Remove deprecated ES option `--es.max-num-spans`

### CODE REVIEW

**Comment by albertteoh** [comment]
Thanks for quickly jumping on this, @BernardTolosajr. We're currently on 1.19.2 and we're targeting this for release 1.21.0 to give folks a chance to migrate over to `--es.max-doc-count`, so I'm not sure if we can merge this change just yet.

I've also restarted Crossdock and Unit Test jobs which appear to be caused by flaky tests.

@BernardTolosajr, I think you'll need to remove the code block here as well: https://github.com/jaegertracing/jaeger/blob/master/plugin/storage/es/options.go#L295

Basically, `grep -ir maxNumSpans` and remove accordingly.

**Comment by BernardTolosajr** [comment]
Noted on this one.

**Review by albertteoh — APPROVED** [review]
lgtm

**Review by yurishkuro — DISMISSED** [review]
I am blocking this for now to prevent being merged accidentally before v1.21

**Comment by albertteoh** [comment]
@BernardTolosajr it's a good time to merge this PR now that we've made a couple of releases. Can you please look at resolving the merge conflict? Let me know if you need help with this. Thanks.

**Comment by yurishkuro** [comment]
fwiw, there are LOTS of other deprecated flags (#2738). But we should not clean them in a single PR since they usually require code changes, better to do one PR per flag.
```
./cmd/flags/admin.go:39:	healthCheckHTTPPortWarning = "(deprecated, will be removed after 2020-03-15 or in release v1.19.0, whichever is later)"
./cmd/flags/admin.go:40:	adminHTTPPortWarning       = "(deprecated, will be removed after 2020-06-30 or in release v1.20.0, whichever is later)"
./cmd/collector/app/builder_flags.go:45:	collectorHTTPPortWarning       = "(deprecated, will be removed after 2020-06-30 or in release v1.20.0, whichever is later)"
./cmd/collector/app/builder_flags.go:46:	collectorGRPCPortWarning       = "(deprecated, will be removed after 2020-06-30 or in release v1.20.0, whichever is later)"
./cmd/collector/app/builder_flags.go:47:	collectorZipkinHTTPPortWarning = "(deprecated, will be removed after 2020-06-30 or in release v1.20.0, whichever is later)"
./cmd/agent/app/reporter/flags.go:51:		flags.String(AgentTagsDeprecated, "", "(deprecated) see --"+agentTags)
./cmd/query/app/flags.go:42:	queryPortWarning        = "(deprecated, will be removed after 2020-08-31 or in release v1.20.0, whichever is later)"
./cmd/query/app/flags.go:43:	queryHOSTPortWarning    = "(deprecated, will be removed after 2020-12-31 or in release v1.21.0, whichever is later)"
./plugin/storage/cassandra/options.go:237:		"(deprecated) Jaeger will automatically detect the version of the dependencies table")
./plugin/storage/cassandra/options.go:241:		"(deprecated) Enable (or disable) host key verification. Use "+nsConfig.namespace+".tls.skip-host-verify instead")
./plugin/storage/es/options.go:186:		"(deprecated, will be removed in release v1.21.0. Please use "+nsConfig.namespace+".max-doc-count). "+
./plugin/storage/es/options_test.go:110:			wantUsage: "(deprecated, will be removed in release v1.21.0. Please use es.max-doc-count). " +
./plugin/storage/es/options_test.go:116:			wantUsage: "(deprecated, will be removed in release v1.21.0. Please use es-archive.max-doc-count). " +
./pkg/config/tlscfg/flags.go:54:		flags.Bool(c.Prefix+tlsEnabledOld, false, "(deprecated) see --"+c.Prefix+tlsEnabled)
./pkg/config/tlscfg/flags.go:69:		flags.Bool(c.Prefix+tlsEnabledOld, false, "(deprecated) see --"+c.Prefix+tlsEnabled)
./pkg/config/tlscfg/flags.go:74:	flags.String(c.Prefix+tlsClientCAOld, "", "(deprecated) see --"+c.Prefix+tlsClientCA)
```

**Comment by BernardTolosajr** [comment]
@albertteoh pushed fixed for the merge conflict. Let me know if i miss something. Thanks

**Comment by BernardTolosajr** [comment]
@albertteoh should i also remove test for MaxNumSpansUsage? 
https://github.com/jaegertracing/jaeger/blob/master/plugin/storage/es/options_test.go#L103

Thanks

**Comment by albertteoh** [comment]
> @albertteoh should i also remove test for MaxNumSpansUsage?
> https://github.com/jaegertracing/jaeger/blob/master/plugin/storage/es/options_test.go#L103

Yes, please @BernardTolosajr.

**Review by yurishkuro — COMMENTED** [review]
please add an entry to changelog as a "breaking change"

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,8 +7,9 @@ Changes by Version
 ### Backend Changes
 
 #### Breaking changes
+* Remove deprecated flags `es.max-num-spans` ([#2482](https://github.com/jaegertracing/jaeger/pull/2482),[@BernardTolosajr](https://github.com/BernardTolosajr))
 
-#### New Features
+### New Features
```
Comment (yurishkuro) [inline CHANGELOG.md:12]: the original level 4 is correct

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,8 +7,9 @@ Changes by Version
 ### Backend Changes
 
 #### Breaking changes
+* Remove deprecated flags `es.max-num-spans` ([#2482](https://github.com/jaegertracing/jaeger/pull/2482),[@BernardTolosajr](https://github.com/BernardTolosajr))
```
Comment (yurishkuro) [inline CHANGELOG.md:10]:
```suggestion
* Remove deprecated flag `--es.max-num-spans`, please use `--es.max-doc-count` ([#2482](https://github.com/jaegertracing/jaeger/pull/2482),[@BernardTolosajr](https://github.com/BernardTolosajr))
```

**Review by yurishkuro — DISMISSED** [review]

**Review by yurishkuro — APPROVED** [review]
@BernardTolosajr Thank you!
